README.md
# Image-Resizer
<p align="center">
  <img width="660" height="500" src="https://i.ibb.co/hyL0Xt0/Image-Resizer.png">
</p>


v1.3/changelog.txt
Here's a changelog highlighting the changes between the two versions of the Python code you provided:

**Changes:**

1. **Image File Paths:**
    - In the first version, image file paths are provided using absolute paths.
    - In the second version, image file paths are provided using relative paths with backslashes (`..\`).

2. **Function Documentation:**
    - In the second version, function documentation (docstrings) has been added for the `choose_image()` and `choose_output_destination()` functions.

3. **Multiple Image Selection:**
    - In the second version, the `choose_image()` function now allows selecting multiple image files at once using the `askopenfilenames` method. The selected file paths are joined with semicolons in the entry field.

4. **Progress Bar:**
    - The second version introduces the `tqdm` library to create a progress bar for the `resize_multiple_images_with_progress()` function.

5. **Multiple Image Resizing:**
    - In the second version, the `resize_multiple_images_with_progress()` function has been introduced to resize multiple selected images with progress tracking.

6. **Dimensions and Format:**
    - In the second version, the predefined dimensions and format choices have been updated.

7. **Custom Dimension Validation:**
    - In the second version, the `resize_multiple_images_with_progress()` function includes validation to ensure that custom width and height inputs are numeric.

8. **Watermark Position:**
    - In the first version, the watermark label is placed at the bottom-right corner.
    - In the second version, the watermark label is placed at the bottom-right corner using the `pack()` method and the `anchor` attribute.

9. **Start GUI Event Loop:**
    - Both versions include the code to start the GUI event loop using the `window.mainloop()` statement.

Please note that this changelog focuses on the changes between the two versions of the provided code snippets. If you have any specific questions about these changes or need further assistance, feel free to ask!

v1.0/ImageResizer.py
import tkinter as tk
from tkinter import filedialog, messagebox
from PIL import Image, ImageTk

# Create a window
window = tk.Tk()
window.title("Image Resizer")
window.geometry("400x250")

# Function to choose an image file
def choose_image():
    file_path = filedialog.askopenfilename(filetypes=[("Image files", "*.png *.jpg *.jpeg")])
    entry.delete(0, tk.END)  # Clear the entry field
    entry.insert(0, file_path)  # Set the chosen file path in the entry field

# Function to choose the output destination
def choose_output_destination():
    folder_path = filedialog.askdirectory()
    output_entry.delete(0, tk.END)  # Clear the entry field
    output_entry.insert(0, folder_path)  # Set the chosen folder path in the entry field

# Function to resize the image
def resize_image():
    # Get the selected file path and output destination from the entry fields
    file_path = entry.get()
    output_folder = output_entry.get()

    try:
        # Open the image using PIL
        image = Image.open(file_path)

        # Get the selected dimension
        selected_dimension = dimension_var.get()
        width, height = dimensions[selected_dimension]

        # Resize the image
        resized_image = image.resize((width, height))

        # Convert the image to RGB color mode
        resized_image = resized_image.convert("RGB")

        # Save the resized image to the output destination
        file_name = file_path.split("/")[-1]  # Extract the file name from the file path
        output_path = f"{output_folder}/{file_name}_resized.jpg"  # Construct the output path
        resized_image.save(output_path)

        # Show a success message box
        messagebox.showinfo("Image Resized", "The image has been successfully resized and saved.")
    except FileNotFoundError:
        messagebox.showerror("Error", "File not found.")
    except PermissionError:
        messagebox.showerror("Error", "Permission denied to access the file.")
    except Exception as e:
        messagebox.showerror("Error", f"Failed to resize the image: {str(e)}")

# Predefined dimension choices
dimensions = {
    "1920 x 1080 pixels": (1920, 1080),
    "1280 x 720 pixels": (1280, 720),
    "250 x 250 pixels": (250, 250),
    "1200 x 630 pixels": (1200, 630)
}

# Create a frame for the image selection
image_frame = tk.Frame(window)
image_frame.pack(pady=10)

# Create a button to choose an image file
choose_button = tk.Button(image_frame, text="Choose Image", command=choose_image, width=15)
choose_button.pack(side=tk.LEFT, padx=(0, 5))

# Create an entry field for the chosen image file path
entry = tk.Entry(image_frame, width=30)
entry.pack(side=tk.LEFT)

# Create a frame for the output destination selection
output_frame = tk.Frame(window)
output_frame.pack(pady=10)

# Create a button to choose the output destination folder
choose_output_button = tk.Button(output_frame, text="Output Destination", command=choose_output_destination, width=15)
choose_output_button.pack(side=tk.LEFT, padx=(0, 5))

# Create an entry field for the chosen output destination folder
output_entry = tk.Entry(output_frame, width=30)
output_entry.pack(side=tk.LEFT)

# Create a frame for the dimension selection
dimension_frame = tk.Frame(window)
dimension_frame.pack(pady=10)

# Create a label for the dimension selection
dimension_label = tk.Label(dimension_frame, text="Choose Dimensions:")
dimension_label.pack(side=tk.LEFT)

# Create a dropdown menu for dimension selection
dimension_var = tk.StringVar(window)
dimension_var.set("1920 x 1080 pixels")  # Set initial dimension
dimension_menu = tk.OptionMenu(dimension_frame, dimension_var, *dimensions.keys())
dimension_menu.pack(side=tk.LEFT, padx=(5, 0))

# Create a button to trigger image resizing
resize_button = tk.Button(window, text="Resize Image", command=resize_image, padx=20, pady=10)
resize_button.pack(pady=20)

# Start the GUI event loop
window.mainloop()


v1.1/ImageResizerVersion1.1.py
import tkinter as tk
from tkinter import filedialog, messagebox
from PIL import Image, ImageTk

# Create a window
window = tk.Tk()
window.title("Image Resizer")
window.geometry("550x500")
window.iconbitmap("resizer.ico")  # Set the path to your icon file

# Load the PNG image
image = Image.open("G:\Software\py\Python Creations\Completed\Projects\ImageResizer\Images\gradient 2.png")
photo = ImageTk.PhotoImage(image)

# Create a label widget to hold the image
background_label = tk.Label(window, image=photo)
background_label.place(x=0, y=0, relwidth=1, relheight=1)

# Add an image label at the top center
# Replace with the actual path to your image file
image_path = "G:\Software\py\Python Creations\Completed\Projects\ImageResizer\Images\Image Resizer.png"
image = Image.open(image_path)
# Replace width and height with your desired dimensions
image = image.resize((130, 120))
image = ImageTk.PhotoImage(image)
image_label = tk.Label(window, image=image)
image_label.pack(pady=(15, 0))


# Function to choose an image file
def choose_image():
    if file_path := filedialog.askopenfilename(
        filetypes=[("Image files", "*.png *.jpg *.jpeg")]
    ):
        entry.delete(0, tk.END)  # Clear the entry field
        entry.insert(0, file_path)  # Set the chosen file path in the entry field


# Function to choose the output destination
def choose_output_destination():
    if folder_path := filedialog.askdirectory():
        output_entry.delete(0, tk.END)  # Clear the entry field
        output_entry.insert(0, folder_path)  # Set the chosen folder path in the entry field


# Function to resize the image
def resize_image():
    # Get the selected file path and output destination from the entry fields
    file_path = entry.get()
    output_folder = output_entry.get()

    try:
        # Open the image using PIL
        image = Image.open(file_path)

        # Get the selected dimension
        selected_dimension = dimension_var.get()
        if selected_dimension == "Custom":
            width = int(custom_width_entry.get())
            height = int(custom_height_entry.get())
        else:
            width, height = dimensions[selected_dimension]

        # Resize the image
        resized_image = image.resize((width, height))

        # Save the resized image to the output destination
        # Extract the file name from the file path
        file_name = file_path.split("/")[-1]
        # Construct the output path with the selected format
        output_path = f"{output_folder}/{file_name}_resized.{format_var.get()}"
        resized_image.save(output_path)

        # Show a success message box
        messagebox.showinfo(
            "Image Resized", "The image has been successfully resized and saved."
        )

        # Display the resized image
        resized_image.show()
    except FileNotFoundError:
        messagebox.showerror("Error", "File not found.")
    except PermissionError:
        messagebox.showerror("Error", "Permission denied to access the file.")
    except Exception as e:
        messagebox.showerror("Error", f"Failed to resize the image: {str(e)}")

# Predefined dimension choices with descriptions
dimensions = {
    "1920 x 1080 (Full HD)": (1920, 1080),  # HD Wallpaper
    "1280 x 720 (HD)": (1280, 720),  # HD Wallpaper
    "400 x 300 (Icon)": (400, 300),  # Icon
    "250 x 250 (Thumbnail)": (250, 250),  # Thumbnail
    "200 x 200 (Favicon)": (200, 200),  # Favicon
}

# Create a frame for the image selection
image_frame = tk.Frame(window)
image_frame.pack(fill="x", expand=True, pady=10)

# Create a button to choose an image file
choose_button = tk.Button(
    image_frame,
    text="Choose Image",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
    command=choose_image,
)
choose_button.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 5))

# Create an entry field for the chosen image file path
entry = tk.Entry(image_frame, width=48)
entry.pack(side=tk.LEFT, fill="both", expand=True, padx=(0, 10))

# Create a frame for the output destination selection
output_frame = tk.Frame(window)
output_frame.pack(fill="x", expand=True, pady=10)

# Create a button to choose the output destination folder
choose_output_button = tk.Button(
    output_frame,
    text="Save Folder",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
    command=choose_output_destination,
)
choose_output_button.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 5))

# Create an entry field for the chosen output destination folder
output_entry = tk.Entry(output_frame, width=45)
output_entry.pack(side=tk.LEFT, fill="both", expand=True, padx=(0, 10))

# Create a frame for the dimension selection
dimension_frame = tk.Frame(window)
dimension_frame.pack(fill="both", expand=True, pady=(10, 30))

# Create a label for the dimension selection
dimension_label = tk.Label(
    dimension_frame,
    text="Choose Dimensions:",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
)
dimension_label.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 0))

# Create a dropdown menu for dimension selection
dimension_var = tk.StringVar(window)
dimension_var.set("1920 x 1080 (Full HD)")  # Set initial dimension
dimension_menu = tk.OptionMenu(
    dimension_frame, dimension_var, *list(dimensions.keys()) + ["Custom"]
)
dimension_menu.pack(side=tk.LEFT, fill="both", expand=True, padx=(5, 10))

# Create a frame for custom dimension input
custom_dimension_frame = tk.Frame(window)

# Create labels and entry fields for custom width and height
custom_width_label = tk.Label(
    custom_dimension_frame,
    text="Custom Width:",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
)
custom_width_label.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 5))

custom_width_entry = tk.Entry(custom_dimension_frame, width=10)
custom_width_entry.pack(side=tk.LEFT, fill="both", expand=True, padx=(0, 5))

custom_height_label = tk.Label(
    custom_dimension_frame,
    text="Custom Height:",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
)
custom_height_label.pack(side=tk.LEFT, fill="both", expand=True, padx=(5, 5))

custom_height_entry = tk.Entry(custom_dimension_frame, width=10)
custom_height_entry.pack(side=tk.LEFT, fill="both", expand=True, padx=(0, 10))


# Add the custom dimension frame to the window
custom_dimension_frame.pack(fill="both", expand=True)

# Function to show or hide the custom dimension frame based on the selected dimension
def toggle_custom_dimension(*args):
    selected_dimension = dimension_var.get()
    if selected_dimension == "Custom":
        custom_dimension_frame.pack(fill="both", expand=True)
    else:
        custom_dimension_frame.pack_forget()

# Bind the toggle_custom_dimension function to the dimension variable
dimension_var.trace("w", toggle_custom_dimension)

# Create a button to resize the image
resize_button = tk.Button(
    window,
    text="Resize Image",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 12, "bold"),
    command=resize_image,
)
resize_button.pack(fill="both", padx=10, pady=10)

# Adjust the width and height as desired
resize_button.config(width=12, height=1)
resize_button.pack(pady=(5, 17))

# Watermark label
watermark_label = tk.Label(
    window, text="pudszTTIOT", font=("Corbel", 9), bg="#2A292B", fg="#46F953"
)
watermark_label.place(relx=1.0, rely=1.0, anchor="se", x=0, y=0)

# Start the GUI event loop
window.mainloop()


v1.2/ImageResizerVersion1.2.py
import tkinter as tk
from tkinter import filedialog, messagebox
from PIL import Image, ImageTk

# Create a window
window = tk.Tk()
window.title("Image Resizer")
window.geometry("550x500")
window.iconbitmap("resizer.ico")  # Set the path to your icon file

# Load the PNG image
image = Image.open("G:\Software\py\Python Creations\Completed\Projects\ImageResizer\Images\gradient 2.png")
photo = ImageTk.PhotoImage(image)

# Create a label widget to hold the image
background_label = tk.Label(window, image=photo)
background_label.place(x=0, y=0, relwidth=1, relheight=1)

# Add an image label at the top center
# Replace with the actual path to your image file
image_path = "G:\Software\py\Python Creations\Completed\Projects\ImageResizer\Images\Image Resizer.png"
image = Image.open(image_path)
# Replace width and height with your desired dimensions
image = image.resize((130, 120))
image = ImageTk.PhotoImage(image)
image_label = tk.Label(window, image=image)
image_label.pack(pady=(15, 0))


# Function to choose an image file
def choose_image():
    if file_path := filedialog.askopenfilename(
        filetypes=[("Image files", "*.png *.jpg *.jpeg")]
    ):
        entry.delete(0, tk.END)  # Clear the entry field
        entry.insert(0, file_path)  # Set the chosen file path in the entry field


# Function to choose the output destination
def choose_output_destination():
    if folder_path := filedialog.askdirectory():
        output_entry.delete(0, tk.END)  # Clear the entry field
        output_entry.insert(0, folder_path)  # Set the chosen folder path in the entry field


# Function to resize the image
def resize_image():
    # Get the selected file path and output destination from the entry fields
    file_path = entry.get()
    output_folder = output_entry.get()

    try:
        # Open the image using PIL
        image = Image.open(file_path)

        # Get the selected dimension
        selected_dimension = dimension_var.get()
        if selected_dimension == "Custom":
            width = int(custom_width_entry.get())
            height = int(custom_height_entry.get())
        else:
            width, height = dimensions[selected_dimension]

        # Resize the image
        resized_image = image.resize((width, height))

        # Save the resized image to the output destination
        # Extract the file name from the file path
        file_name = file_path.split("/")[-1]
        # Construct the output path with the selected format
        output_path = f"{output_folder}/{file_name}_resized.{format_var.get()}"
        resized_image.save(output_path)

        # Show a success message box
        messagebox.showinfo(
            "Image Resized", "The image has been successfully resized and saved."
        )

        # Display the resized image
        resized_image.show()
    except FileNotFoundError:
        messagebox.showerror("Error", "File not found.")
    except PermissionError:
        messagebox.showerror("Error", "Permission denied to access the file.")
    except Exception as e:
        messagebox.showerror("Error", f"Failed to resize the image: {str(e)}")

# Predefined dimension choices with descriptions
dimensions = {
    "1920 x 1080 (Full HD)": (1920, 1080),  # HD Wallpaper
    "1280 x 720 (HD)": (1280, 720),  # HD Wallpaper
    "400 x 300 (Icon)": (400, 300),  # Icon
    "250 x 250 (Thumbnail)": (250, 250),  # Thumbnail
    "200 x 200 (Favicon)": (200, 200),  # Favicon
}

# Create a frame for the image selection
image_frame = tk.Frame(window)
image_frame.pack(fill="x", expand=True, pady=10)

# Create a button to choose an image file
choose_button = tk.Button(
    image_frame,
    text="Choose Image",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
    command=choose_image,
)
choose_button.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 5))

# Create an entry field for the chosen image file path
entry = tk.Entry(image_frame, width=48)
entry.pack(side=tk.LEFT, fill="both", expand=True, padx=(0, 10))

# Create a frame for the output destination selection
output_frame = tk.Frame(window)
output_frame.pack(fill="x", expand=True, pady=10)

# Create a button to choose the output destination folder
choose_output_button = tk.Button(
    output_frame,
    text="Save Folder",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
    command=choose_output_destination,
)
choose_output_button.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 5))

# Create an entry field for the chosen output destination folder
output_entry = tk.Entry(output_frame, width=45)
output_entry.pack(side=tk.LEFT, fill="both", expand=True, padx=(0, 10))

# Create a frame for the dimension selection
dimension_frame = tk.Frame(window)
dimension_frame.pack(fill="both", expand=True, pady=(10, 30))

# Create a label for the dimension selection
dimension_label = tk.Label(
    dimension_frame,
    text="Choose Dimensions:",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
)
dimension_label.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 0))

# Create a dropdown menu for dimension selection
dimension_var = tk.StringVar(window)
dimension_var.set("1920 x 1080 (Full HD)")  # Set initial dimension
dimension_menu = tk.OptionMenu(
    dimension_frame, dimension_var, *list(dimensions.keys()) + ["Custom"]
)
dimension_menu.pack(side=tk.LEFT, fill="both", expand=True, padx=(5, 10))

# Create a frame for custom dimension input
custom_dimension_frame = tk.Frame(window)

# Create labels and entry fields for custom width and height
custom_width_label = tk.Label(
    custom_dimension_frame,
    text="Custom Width:",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
)
custom_width_label.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 5))

custom_width_entry = tk.Entry(custom_dimension_frame, width=10)
custom_width_entry.pack(side=tk.LEFT, fill="both", expand=True, padx=(0, 5))

custom_height_label = tk.Label(
    custom_dimension_frame,
    text="Custom Height:",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
)
custom_height_label.pack(side=tk.LEFT, fill="both", expand=True, padx=(5, 5))

custom_height_entry = tk.Entry(custom_dimension_frame, width=10)
custom_height_entry.pack(side=tk.LEFT, fill="both", expand=True, padx=(0, 10))

# Create a frame for format selection
format_frame = tk.Frame(window)
format_frame.pack(fill="both", expand=True, pady=(10, 30))

# Create a label for the format selection
format_label = tk.Label(
    format_frame,
    text="Choose Format:",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
)
format_label.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 0))

# Create a dropdown menu for format selection
format_var = tk.StringVar(window)
format_var.set("png")  # Set initial format
format_menu = tk.OptionMenu(format_frame, format_var, "gif", "ico", "jpeg", "jpg", "png")
format_menu.pack(side=tk.LEFT, fill="both", expand=True, padx=(5, 10))

# Add the custom dimension frame to the window
custom_dimension_frame.pack(fill="both", expand=True)

# Function to show or hide the custom dimension frame based on the selected dimension
def toggle_custom_dimension(*args):
    selected_dimension = dimension_var.get()
    if selected_dimension == "Custom":
        custom_dimension_frame.pack(fill="both", expand=True)
    else:
        custom_dimension_frame.pack_forget()

# Bind the toggle_custom_dimension function to the dimension variable
dimension_var.trace("w", toggle_custom_dimension)

# Create a button to resize the image
resize_button = tk.Button(
    window,
    text="Resize Image",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 12, "bold"),
    command=resize_image,
)
resize_button.pack(fill="both", padx=10, pady=10)

# Adjust the width and height as desired
resize_button.config(width=12, height=1)
resize_button.pack(pady=(5, 17))

# Watermark label
watermark_label = tk.Label(
    window, text="pudszTTIOT", font=("Corbel", 9), bg="#2A292B", fg="#46F953"
)
watermark_label.place(relx=1.0, rely=1.0, anchor="se", x=0, y=0)

# Start the GUI event loop
window.mainloop()


v1.3/ImageResizerVersion1.3.py
import tkinter as tk
from tkinter import filedialog, messagebox
from PIL import Image, ImageTk

# Create a window
window = tk.Tk()
window.title("Image Resizer")
window.geometry("550x500")
window.iconbitmap("resizer.ico")  # Set the path to your icon file

# Load the PNG image
image = Image.open("..\Images\gradient 2.png")
photo = ImageTk.PhotoImage(image)

# Create a label widget to hold the image
background_label = tk.Label(window, image=photo)
background_label.place(x=0, y=0, relwidth=1, relheight=1)

# Add an image label at the top center
# Replace with the actual path to your image file
image_path = r"..\Images\Image Resizer.png"
image = Image.open(image_path)
# Replace width and height with your desired dimensions
image = image.resize((130, 120))
image = ImageTk.PhotoImage(image)
image_label = tk.Label(window, image=image)
image_label.pack(pady=(15, 0))

def choose_image():
    """Open a file dialog to choose multiple image files."""
    file_paths = filedialog.askopenfilenames(filetypes=[("Image files", "*.png *.jpg *.jpeg")])
    if file_paths:
        entry.delete(0, tk.END)
        entry.insert(0, ";".join(file_paths))  # Join the selected file paths with semicolon

# Function to choose the output destination
def choose_output_destination():
    """Open a folder dialog to choose the output destination folder."""
    folder_path = filedialog.askdirectory()
    if folder_path:
        output_entry.delete(0, tk.END)  # Clear the entry field
        output_entry.insert(0, folder_path)  # Set the chosen folder path in the entry field

from tqdm import tqdm  # Import the tqdm library

# Function to resize multiple images with a progress bar
def resize_multiple_images_with_progress():
    """Resize multiple selected images and save them to the output destination with a progress bar."""
    file_paths = entry.get().split(';')
    output_folder = output_entry.get()

    try:
        with tqdm(total=len(file_paths), desc="Resizing Images", unit="image") as progress_bar:
            for file_path in file_paths:
                image = Image.open(file_path)

                selected_dimension = dimension_var.get()
                if selected_dimension == "Custom":
                    custom_width = custom_width_entry.get()
                    custom_height = custom_height_entry.get()

                    if not custom_width.isdigit() or not custom_height.isdigit():
                        raise ValueError("Custom width and height must be numeric.")

                    width = int(custom_width)
                    height = int(custom_height)
                else:
                    width, height = dimensions[selected_dimension]

                resized_image = image.resize((width, height))

                file_name = file_path.split("/")[-1]
                output_path = f"{output_folder}/{file_name}_resized.{format_var.get()}"
                resized_image.save(output_path)

                progress_bar.update(1)

        messagebox.showinfo("Images Resized", "All images have been successfully resized and saved.")
    except Exception as e:
        messagebox.showerror("Error", "An error occurred: {}".format(str(e)))


# Predefined dimension choices with descriptions
dimensions = {
    "3000 x 3000 (Album Cover Art)": (3000, 3000),  # Album Cover Art
    "1920 x 1080 (Full HD)": (1920, 1080),  # HD Wallpaper
    "1280 x 720 (HD)": (1280, 720),  # HD Wallpaper
    "256 x 256 (Icon)": (256, 256),  # Icon
    "16 x 16 (Favicon)": (16, 16),  # Favicon
}

# Create a frame for the image selection
image_frame = tk.Frame(window)
image_frame.pack(fill="x", expand=True, pady=10)

# Create a button to choose an image file
choose_button = tk.Button(
    image_frame,
    text="Choose Image",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
    command=choose_image,
)
choose_button.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 5))

# Create an entry field for the chosen image file path
entry = tk.Entry(image_frame, width=48)
entry.pack(side=tk.LEFT, fill="both", expand=True, padx=(0, 10))

# Create a frame for the output destination selection
output_frame = tk.Frame(window)
output_frame.pack(fill="x", expand=True, pady=10)

# Create a button to choose the output destination folder
choose_output_button = tk.Button(
    output_frame,
    text="Save Folder",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
    command=choose_output_destination,
)
choose_output_button.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 5))

# Create an entry field for the chosen output destination folder
output_entry = tk.Entry(output_frame, width=45)
output_entry.pack(side=tk.LEFT, fill="both", expand=True, padx=(0, 10))

# Create a frame for the dimension selection
dimension_frame = tk.Frame(window)
dimension_frame.pack(fill="both", expand=True, pady=(10, 30))

# Create a label for the dimension selection
dimension_label = tk.Label(
    dimension_frame,
    text="Choose Dimensions:",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
)
dimension_label.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 0))

# Create a dropdown menu for dimension selection
dimension_var = tk.StringVar(window)
dimension_var.set("3000 x 3000 (Album Cover Art)")  # Set initial dimension
dimension_menu = tk.OptionMenu(
    dimension_frame, dimension_var, *list(dimensions.keys()) + ["Custom"]
)
dimension_menu.pack(side=tk.LEFT, fill="both", expand=True, padx=(5, 10))

# Create a frame for custom dimension input
custom_dimension_frame = tk.Frame(window)

# Create labels and entry fields for custom width and height
custom_width_label = tk.Label(
    custom_dimension_frame,
    text="Custom Width:",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
)
custom_width_label.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 5))

custom_width_entry = tk.Entry(custom_dimension_frame, width=10)
custom_width_entry.pack(side=tk.LEFT, fill="both", expand=True, padx=(0, 5))

custom_height_label = tk.Label(
    custom_dimension_frame,
    text="Custom Height:",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
)
custom_height_label.pack(side=tk.LEFT, fill="both", expand=True, padx=(5, 5))

custom_height_entry = tk.Entry(custom_dimension_frame, width=10)
custom_height_entry.pack(side=tk.LEFT, fill="both", expand=True, padx=(0, 10))

# Create a frame for format selection
format_frame = tk.Frame(window)
format_frame.pack(fill="both", expand=True, pady=(10, 30))

# Create a label for the format selection
format_label = tk.Label(
    format_frame,
    text="Choose Format:",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 11),
)
format_label.pack(side=tk.LEFT, fill="both", expand=True, padx=(10, 0))

# Create a dropdown menu for format selection
format_var = tk.StringVar(window)
format_var.set("png")  # Set initial format
format_menu = tk.OptionMenu(format_frame, format_var, "gif", "ico", "jpeg", "jpg", "png")
format_menu.pack(side=tk.LEFT, fill="both", expand=True, padx=(5, 10))

# Add the custom dimension frame to the window
custom_dimension_frame.pack(fill="both", expand=True)

# Function to show or hide the custom dimension frame based on the selected dimension
def toggle_custom_dimension(*args):
    """Toggle the visibility of the custom dimension frame based on the selected dimension."""
    selected_dimension = dimension_var.get()
    if selected_dimension == "Custom":
        custom_dimension_frame.pack(fill="both", expand=True)
    else:
        custom_dimension_frame.pack_forget()

# Bind the toggle_custom_dimension function to the dimension variable
dimension_var.trace("w", toggle_custom_dimension)

# Create a button to resize the images with a progress bar
resize_button = tk.Button(
    window,
    text="Resize Images",
    bg="#E7E9EA",
    fg="#20211A",
    justify="center",
    font=("Tahoma", 12, "bold"),
    command=resize_multiple_images_with_progress,
)
resize_button.pack(fill="both", padx=10, pady=10)


# Adjust the width and height as desired
resize_button.config(width=12, height=1)
resize_button.pack(pady=(5, 17))

# Watermark label
watermark_label = tk.Label(
    window, text="pudszTTIOT", font=("Corbel", 9), bg="#2A292B", fg="#46F953"
)
watermark_label.pack(side=tk.BOTTOM, anchor="e")

# Start the GUI event loop
window.mainloop()
